### PR TITLE
fix issue 227 and 228

### PR DESCRIPTION
**Pull request by londumas**
fix issue 227 and 228:
https://github.com/igmhub/picca/issues/228
https://github.com/igmhub/picca/issues/227

do_cf1d computes only one time the correlation and works for the cross corelation

Here is the SiIV x CIV 1d cross correlation

capture d ecran de 2017-11-08 12-03-52

### CODE REVIEW

**Comment by londumas** [comment]
still some changes to be done

**Comment by londumas** [comment]
The 1d auto and the 1d cross LYA give the same results:
in green the cross and in blue the auto
capture d ecran de 2017-11-08 13-40-53

**Comment by londumas** [comment]
The only thing is that for the cross of different forests, the diagonal is zero, so all the 1d arrays are zero "v1d','wv1d','nv1d','c1d','nc1d','nb1d".
The data are then only in the 2d arrays.

Here is an example of the SIIV_CIV 1d cross correlation
capture d ecran de 2017-11-08 13-46-59

**Comment by londumas** [comment]
ready to be merged, can you review @vserret

**Comment by londumas** [comment]
@ngbusca, what are the lines I should put back ?

**Comment by ngbusca** [comment]
@londumas the map should be a pool.map to use multiprocessing. Otherwise it runs on a single thread. So you need to uncomment the definition of pool = Pool(...) and move it to before the if x_correlation block. The replace the two map calls by pool.map. Please remove any commented out lines as well. Please test again before pushing that you get the same results as above.

**Comment by ngbusca** [comment]
@londumas thanks for catching this.

**Comment by londumas** [comment]
@ngbusca, how does this look like ?
I tested and it still gives the same results.
If ok, you can merge.

File: bin/do_cf1d.py [hunk 1]
```diff
@@ -139,6 +140,8 @@ def cf1d(p):
         cfs = map(cf1d,keys)
     else: cfs = map(cf1d,data.keys())
```
Comment (ngbusca) [inline bin/do_cf1d.py]: @londumas you still need to replace map() by pool.map() otherwise it doesn't change anything ;^)

Comment (londumas) [reply bin/do_cf1d.py]: just did and indeed now it multithread

Comment (ngbusca) [reply bin/do_cf1d.py]: still getting the same results?

**Comment by londumas** [comment]
yes, still getting the same results, and I did test it.

**Pull request merged**